index.js

```diff
@@ -1,2 +1,37 @@
+/* eslint-disable func-names, no-extra-bind */
+
 'use strict';
-module.exports = (fn, name) => Object.defineProperty(fn, 'name', {value: name, configurable: true});
+module.exports = (fn, name) => {
+	Object.defineProperty(fn, 'name', {value: name, configurable: true});
+	const functionString = fn.toString();
+	if (
```

Comment:
The whole replacement should only be done lazily when `.toString()` is called on the renamed function.

Reply:
Do you mean the following?:

```diff
@@ -3,7 +3,6 @@
 'use strict';
 module.exports = (fn, name) => {
 	Object.defineProperty(fn, 'name', {value: name, configurable: true});
-	const functionString = fn.toString();
 	if (
 		!(
 			!/^(?:async\s*)?(?:(\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*(?:function)?\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*\*?.*?\s*(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n)\s*)*\(/.test(functionString) ||
@@ -16,11 +15,13 @@ module.exports = (fn, name) => {
 		functionString.indexOf('class') !== 0 &&
 		(!/^(get|set)/.test(functionString) || !functionString.slice(fn.name.length).replace(/\s/g, '').replace(/(\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*/g, '').trim().startsWith('('))
 	) {
+		const functionString = fn.toString();
 		const captured = /^((?:get|set)?(?:async\s*)?(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*(?:function)?\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*\*?\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*)([^]*?)\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*\(/.exec(functionString);
 		fn.toString = (function toString() {
 			return functionString.slice(0, captured[1].length) + name + functionString.slice(captured[1].length + captured[2].length);
 		}).bind(fn);
 	} else if (functionString.indexOf('class') === 0) {
+		const functionString = fn.toString();
 		const captured = /^(class\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*)([^]*?)\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*(?:extends|{)/.exec(functionString);
 		if (captured[2] !== '') {
 			fn.toString = (function toString() {
@@ -28,6 +29,7 @@ module.exports = (fn, name) => {
 			}).bind(fn);
 		}
 	} else if (/^(get|set)/.test(functionString)) {
+		const functionString = fn.toString();
 		fn.toString = (function toString() {
 			return name + functionString.slice(fn.name.length);
 		}).bind(fn);
```

Reply:
No, I mean when the **user** calls `.toString()`.

Reply:
ae44a71

Reply:
Do you mean https://github.com/sindresorhus/rename-fn/pull/3#issuecomment-762419071?